iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateFeederBay.java

```diff
@@ -0,0 +1,189 @@
+ * busbar section should be described in node/breaker topology. The injection is created and connected to
+ * the busbar section with a breaker and a closed disconnector. The injection is also connected to all
+ * the parallel busbar sections, if any, with an open disconnector.
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+public class CreateFeederBay implements NetworkModification {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateFeederBay.class);
+
+    private final InjectionAdder<?> injectionAdder;
+    private final String bbsId;
+    private final int injectionPositionOrder;
+    private final ConnectablePosition.Direction injectionDirection;
+
+    /**
+     * Constructor.
+     *
+     * @param injectionAdder         The injection adder.
+     * @param bbsId                  The ID of the existing busbar section where we want to connect the injection.
+     *                               Please note that there will be switches between this busbar section and the connection point of the injection. This switch will be closed.
+     * @param injectionPositionOrder The order of the injection to be attached from its extension {@link ConnectablePosition}.
+     * @param injectionDirection     The direction of the injection to be attached from its extension {@link ConnectablePosition}.
+     */
+    public CreateFeederBay(InjectionAdder<?> injectionAdder, String bbsId, int injectionPositionOrder, ConnectablePosition.Direction injectionDirection) {
+        this.injectionAdder = Objects.requireNonNull(injectionAdder);
+        this.bbsId = Objects.requireNonNull(bbsId);
+        this.injectionPositionOrder = injectionPositionOrder;
+        this.injectionDirection = Objects.requireNonNull(injectionDirection);
+    }
+
+    public CreateFeederBay(InjectionAdder<?> injectionAdder, String bbsId, int injectionPositionOrder) {
+        this(injectionAdder, bbsId, injectionPositionOrder, ConnectablePosition.Direction.BOTTOM);
+    }
+
+    @Override
+    public void apply(Network network, ComputationManager computationManager) {
+        apply(network);
+    }
+
+    @Override
+    public void apply(Network network) {
+        apply(network, false, Reporter.NO_OP);
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, Reporter reporter) {
+        BusbarSection bbs = network.getBusbarSection(bbsId);
+        if (bbs == null) {
+            LOGGER.error("Busbar section {} not found.", bbsId);
+            notFoundBusbarSectionReport(reporter, bbsId);
+            if (throwException) {
+                throw new PowsyblException(String.format("Busbar section %s not found.", bbsId));
+            }
+            return;
+        }
+
+        VoltageLevel voltageLevel = bbs.getTerminal().getVoltageLevel();
+        TopologyKind topologyKind = voltageLevel.getTopologyKind();
+        if (topologyKind != TopologyKind.NODE_BREAKER) {
```

Comment:
Am I wrong or we can remove this check? A busbar section makes no sense if the voltage level is not described at node/breaker level?